src/crimson/osd/osd_operations/pg_splitting.cc

```diff
@@ -53,11 +53,9 @@ seastar::future<> PGSplitting::start()
     auto pg_epoch = child_pg_info.second;
     children_pgids.insert(child_pg);
     // Map each child pg ID to a core
-    return shard_services.get_or_create_pg(child_pg).then([FNAME, child_pg]
-      (auto core) {
-      DEBUG(" PG {} mapped to {}", child_pg.pgid, core);
-      return seastar::now();
-    }).then([this, FNAME, child_pg, pg_epoch] {
+    return shard_services.create_split_pg_mapping(child_pg, seastar::this_shard_id()).then(
```

Comment:
Following last comment, by using the existing `PGShardMapping` interface, would this work?

`shard_services.get_pg_to_shard_mapping().get_or_create_pg_mapping(child_pg, seastar::this_shard_id())`

Reply:
That would be cleaner, I'll look into it thanks!